src/contrib/mlir/compiler/lowerer.cpp

```diff
@@ -842,6 +837,22 @@ namespace
             // image spatial bounds
             imgSpatialLbs.push_back(vImages.lb(i + 2));
             imgSpatialUbs.push_back(vImages.ub(i + 2));
+
+            // Check if we have any padding and collect pad values
+            IntegerAttr iAttr = padBelow[i].cast<IntegerAttr>();
+            int padValue = iAttr.getInt();
+            if (padValue)
+            {
+                withPadding = true;
+            }
+            padBelowIntValues.push_back(padValue);
+
+            iAttr = padAbove[i].cast<IntegerAttr>();
+            padValue = iAttr.getInt();
+            if (padValue)
+            {
+                withPadding = true;
+            }
```

Comment:
For my understanding... we don't need `padAboveValues`?

Reply:
Not to the check if we are inside the non-padded range. PadAbove is actually included in the inferred output shape early on in nGraph, so the loop bounds are adjusted here to include padAbove though.